### PR TITLE
HTTP Status Codes Cheatsheet

### PR DESCRIPTION
**Pull request by ShreyasMinocha**
**What does your Instant Answer do?**
This IA shows response/status codes returned by the HTTP.

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://www.cheatography.com/kstep/cheat-sheets/http-status-codes/

**Why did you choose this data source?**
It shows a simplified reference to most HTTP status codes.

**Are there any other alternative (better) data sources?**
No.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Network admins, people using proxies.

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
No.

**Are you having any problems? Do you need our help with anything?**
No.

-----
IA Page: https://duck.co/ia/view/http_status_codes_cheat_sheet

### CODE REVIEW

**Comment by seisfeld** [comment]
Feel free to have a look at #1423 where i did the same for "SIP response codes". Maybe it makes sense to align yours in the form and triggers, since they are very similar. Gives a consistent look and feel. What do you guys think?

**Comment by ShreyasMinocha** [comment]
seisfeld. Sure. I'll look into it when I get time. I don't however feel the need to give a detailled summary for each status code.

**Comment by seisfeld** [comment]
@ShreyasMinocha Sure, whatever you feel is needed. :smile: It's just that i discussed the naming with @moollaza when i submitted my  #1423. In the end it's his call not mine hehe. I'm sure he will pick up this one soon.

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 1]
```diff
@@ -0,0 +1,236 @@
+{
+    "id": "http_status_codes_cheatsheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: Please update to `http_status_codes_cheat_sheet`

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    "name": "HTTP Status Codes",
+    "description": "Informational, Success and Error Codes returned by HTTP",
+    "metadata": {
+        "sourceName": "Cheatography",
+        "sourceUrl": "http://www.cheatography.com/kstep/cheat-sheets/http-status-codes/"
+    },
+    "template_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: Let's use the `reference` template here

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    "section_order": [
+        "HTTP Informational Codes (1xx)",
+        "HTTP Successful Codes (2xx)",
+        "HTTP Redirection Codes (3xx)",
+        "HTTP Client Error Codes (4xx)",
+        "HTTP Server Error Codes (5xx)"
+
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: Please remove this empty line

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            "val": "Multi Status",
+            "key": "207"
+        }, {
+            "val": "Already Reported",
+            "key": "208"
+        }, {
+            "val": "IM used",
+            "key": "226"
+        }],
+"HTTP Redirection Codes (3xx)": [{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: Whoops, indentation looks a little off here

**Comment by moollaza** [comment]
@seisfeld thanks a lot for your input! I'm happy to see some cross-discussion happening.

>In the end it's his call not mine hehe. I'm sure he will pick up this one soon.

Ultimately, it's not mine either! It's the community's! If you guys want to see more info, then it doesn't hurt to show it. I feel like this is OK in its current form, but personally, I would like to see some information given about each status codes. I guess it depends on the use case though because if I know what I'm looking for, it's likely that I'll actually search "HTTP 502" instead of lookup the cheat sheet. Eventually though, we'll be making the content searchable so we can surface cheat sheets in a more organic fashion.

@ShreyasMinocha @seisfeld if you guys are OK with this, I'm happy to merged one my comments have been addressed

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 5]
```diff
@@ -0,0 +1,236 @@
+{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: please change the filename to `http-status-codes.json`

File: share/goodie/cheat_sheets/json/http_status_codes.json [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    "id": "http_status_codes_cheatsheet",
+    "name": "HTTP Status Codes",
+    "description": "Informational, Success and Error Codes returned by HTTP",
+    "metadata": {
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: Let's add some aliases: `"aliases": ["http status code", "http error code", "http error codes"],`

**Comment by seisfeld** [comment]
@moollaza Appreciate your open discussion. :smile: Mine was called sip-response-codes.json, so i'd either rename this http-response-codes.json or at least alias "http response codes". I had "sip status codes" as alias. Reason i'm mentioning all this it the fact that these status codes are very similar (SIP ist based on HTTP) so searching and finding these should be similar as well.

File: share/goodie/cheat_sheets/json/http_status_codes.json (hunk 6 above)
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/http_status_codes.json]: I'd also add "http response codes", "http response code", "http response status codes", "http response status code"

**Comment by jagtalon** [comment]
@ShreyasMinocha I think I'd like more descriptive values for the status codes like in http://httpstatus.es/. What do you think @moollaza?

**Comment by mintsoft** [comment]
@jagtalon I agree with the more descriptive values, as long as the HTTP literal (i.e `100 Continue`, `404 Not Found`) is on the cheatsheet

**Comment by ShreyasMinocha** [comment]
@moollaza @seisfeld @jagtalon @mintsoft  Thanks for the inputs. I'll make these changes within the week.

**Comment by moollaza** [comment]
@ShreyasMinocha this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@ShreyasMinocha congrats! It's live: https://duckduckgo.com/?q=http+status+codes+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**